editor/src/utils/canvas-react-utils.ts

```diff
@@ -119,28 +120,30 @@ function shouldIncludeDataUID(type: any): boolean {
 
 function attachDataUidToRoot(
   originalResponse: React.ReactElement | null | undefined,
-  dataUid: string | null,
+  dataUids: string | null,
 ): React.ReactElement | null
 function attachDataUidToRoot(
   originalResponse: Array<React.ReactElement | null>,
-  dataUid: string | null,
+  dataUids: string | null,
 ): Array<React.ReactElement | null>
 function attachDataUidToRoot(
   originalResponse: React.ReactElement | Array<React.ReactElement | null> | null | undefined,
-  dataUid: string | null,
+  dataUids: string | null,
 ): React.ReactElement | Array<React.ReactElement | null> | null {
-  if (originalResponse == null || dataUid == null) {
+  if (originalResponse == null || dataUids == null) {
     return originalResponse as any
   } else if (Array.isArray(originalResponse)) {
     // the response was an array of elements
-    return originalResponse.map((element) => attachDataUidToRoot(element, dataUid))
+    return originalResponse.map((element) => attachDataUidToRoot(element, dataUids))
   } else if (!React.isValidElement(originalResponse as any)) {
     return originalResponse
   } else {
     if (shouldIncludeDataUID(originalResponse.type)) {
-      return React.cloneElement(originalResponse, { [UTOPIA_UID_KEY]: dataUid })
+      return React.cloneElement(originalResponse, {
+        [UTOPIA_UIDS_KEY]: appendToUidString(originalResponse.props[UTOPIA_UIDS_KEY], dataUids),
+      })
     } else {
-      return React.cloneElement(originalResponse)
+      return React.cloneElement(originalResponse) // This is probably an unnecessary clone
```

Comment:
Good catch. It looks like it is (in other cases we're returning `originalResponse`)

Reply:
I'll take this